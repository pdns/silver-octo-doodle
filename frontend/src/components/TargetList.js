import React from 'react';
import PropTypes from 'prop-types';
import List from '@material-ui/core/List';
import { graphql, createFragmentContainer } from 'react-relay';
import TargetListItem from './TargetListItem';


function TargetList(props) {
  const items = props.host.targets.edges.map(({ node }) =>
    <TargetListItem key={node.id} target={node} />);

  return (
    <List component="div" disablePadding>
      {items}
    </List>
  );
}

TargetList.propTypes = {
  host: PropTypes.object,
};

export default createFragmentContainer(
  TargetList,
  graphql`
    fragment TargetList_host on Host {
      targets(first: 2147483647) @connection(key: "HostListItem_targets") {
        edges {
          node {
            id
            ...TargetListItem_target
          }
        }
      }
    }
  `,
);
