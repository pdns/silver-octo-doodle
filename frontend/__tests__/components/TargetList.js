/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import TargetList from '../../src/components/TargetList';

jest.mock('react-relay', () => ({
  createFragmentContainer: x => x,
}));

jest.mock('@material-ui/core/List', () => 'List');

jest.mock('../../src/components/TargetListItem', () => 'TargetListItem');

describe('TargetList', () => {
  it('should render correctly', () => {
    const host = {
      targets: {
        edges: [
          { node: { id: '1' } },
          { node: { id: '2' } },
        ],
      },
    };
    const tree = renderer
      .create(<TargetList host={host} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
