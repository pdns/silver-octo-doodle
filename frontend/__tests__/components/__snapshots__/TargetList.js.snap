// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TargetList should render correctly 1`] = `
<List
  component="div"
  disablePadding={true}
>
  <TargetListItem
    target={
      Object {
        "id": "1",
      }
    }
  />
  <TargetListItem
    target={
      Object {
        "id": "2",
      }
    }
  />
</List>
`;
